### DISCUSSION TITLE
Processing geonames.org data to load csv files to kuzu

### DISCUSSION THREAD

**Discussion by uwegeercken** [kuzudb/kuzu#1340] [Show and tell]
Hi everybody,

I have designed an ETL pipeline to process data from geonames.org and load the resulting csv files into kuzu.

What is geonames? This is from their homepage:
Quote: "The GeoNames geographical database covers all countries and contains over eleven million placenames that are available for  download free of charge."

They have free data to download, including files on countries, administrative divisions, feature codes ( like e.g. river, hill, mine, populated place, etc) etc.

The ETL pipeline processes the data and creates nodes and relations between:

- geographical places = 11842832
- admin divisions = 515024
- countries = 193
- feature codes = 680
- relation geographical places to admin division (multiple per place) = 21863415
- relation geographical places to countries = 11734891
- relation geographical places to feature codes = 11842832
 
 In future I plan to integrate more data - like continents, timezones, etc. I also want to time the performance of the data import and some cypher queries against the data set. 

My project is on github: https://github.com/uwegeercken/tweakstreet-kuzudb

Is anybody interested in something like this? Or maybe somebody wants to help to enhanced it?

Greetings,

Uwe

**Comment by semihsalihoglu-uw** [comment]
Hi Uwe,

I assume this is a general call to other users who might be trying Kuzu out, right? The core team has a very busy schedule developing the system. You can also join our [Slack workspace](https://join.slack.com/t/kuzudb/shared_invite/zt-1n67h736q-E3AFGSI4w~ljlFMYr3_Sjg), and ask there. It is currently quite small but as we grow a user community, we want Slack to be a place users discuss Kuzu-related topics.

Best,

Semih

**Comment by uwegeercken** [comment]
I was writing a comment here as for the section it says "Show off something you've made". But I will jump over to Slack and have a look there. Thanks.

Uwe

**Comment by uwegeercken** [comment]
I need an invitation to the slack channel. Can you send it for uwe.geercken@web.de?

**Reply by ray6080** [reply]
Sure! @uwegeercken We've send the invitation to your email.

**Reply by uwegeercken** [reply]
Many thanks

**Comment by prrao87** [comment]
We're now showcasing these kinds of tools/add-ons in our Discord server, on the "show-and-tell" channel. Join us there!
https://kuzudb.com/chat
